kernel/src/table_changes/log_replay.rs

```diff
@@ -156,20 +156,22 @@ impl LogReplayScanner {
         // vectors are resolved so that we can skip both actions in the pair.
         let action_iter = engine.get_json_handler().read_json_files(
             &[commit_file.location.clone()],
-            visitor_schema,
+            PreparePhaseVisitor::schema(),
             None, // not safe to apply data skipping yet
         )?;
 
         let mut remove_dvs = HashMap::default();
         let mut add_paths = HashSet::default();
         let mut has_cdc_action = false;
+        let mut timestamp = commit_file.location.last_modified;
         for actions in action_iter {
             let actions = actions?;
 
             let mut visitor = PreparePhaseVisitor {
                 add_paths: &mut add_paths,
                 remove_dvs: &mut remove_dvs,
                 has_cdc_action: &mut has_cdc_action,
+                commit_timestamp: &mut timestamp,
```

Comment:
would this be clearer?
```suggestion
                in_commit_timestamp: &mut timestamp,
```

Reply:
We initialize this field with the file modification timestamp, so it would be inaccurate to call it that. I do like the update you made below tho when we actually read ICT from a commitinfo.